### PR TITLE
Add support for image pull secrets

### PR DESCRIPTION
**Pull request by 0marq**
Hello, 

@ForePaaS, we would like to use your project to pull images from private repositories. I took the liberty of forking your project and adding support for Kubernetes Image Pull Secrets.

Please let me know what you think of my changes. 

Regards, 

Omar

### CODE REVIEW

File: deploy/helm/values.yaml [hunk 1]
```diff
@@ -16,4 +16,4 @@ configMap:
   cachingCpuRequest: ".05"
   cachingCpuLimit: ".2"
   nodeSelector: "{}"
-
+  imagePullSecrets: ""
```
Comment (ibuziuk) [inline deploy/helm/values.yaml:19]: No new line in the end

File: README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ The config values to be set are:
 | `NAMESPACE`              | Namespace where daemonset is to be created | `kubernetes-image-puller` |
 | `IMAGES`                 | List of images to be cached, in the format `<name>=<image>;...` | Contains a default list of images, but should be configured when deploying |
 | `NODE_SELECTOR` | Node selector applied to pods created by the daemonset       | `'{}'` |
+| `IMAGE_PULL_SECRETS` | List of image pull secrets, in the format `pullsecret1;...` to add to pods created by the daemonset       | `""` |
```
Comment (ibuziuk) [inline README.md:28]: is it expected that the `pullsecret1` is created by the cluster admin? should it be created in the very same namespace where the image puller is installed?

File: deploy/openshift/configmap.yaml [hunk 3]
```diff
@@ -42,3 +43,5 @@ parameters:
   value: ".2"
 - name: NODE_SELECTOR
   value: "{}"
+- name: IMAGE_PULL_SECRETS
+  value: ""
```
Comment (ibuziuk) [inline deploy/openshift/configmap.yaml:47]: no new line in the end

**Review by ibuziuk — CHANGES_REQUESTED** [review]
@0marq thank you very much for the contribution. The PR looks really good - could you please address the comments and provide the PR for documentation https://www.eclipse.org/che/docs/che-7/administration-guide/caching-images-for-faster-workspace-start/

The docs repo - https://github.com/eclipse/che-docs

File: deploy/helm/values.yaml (hunk 1 above)
Comment (0marq) [reply deploy/helm/values.yaml:19]: There actually is a new line in the end here ...

File: deploy/openshift/configmap.yaml (hunk 3 above)
Comment (0marq) [reply deploy/openshift/configmap.yaml]: Added a new line

File: README.md (hunk 2 above)
Comment (0marq) [reply README.md:28]: Yep exactly, secrets are required to be in the same namespace and to be created by the admin

**Comment by 0marq** [comment]
@ibuziuk Thank you for the review, as requested I've update the docs in the PR https://github.com/eclipse/che-docs/pull/1957
I've also responded to your comments, feel free if there is anything else you need me to do ! :)

File: deploy/helm/values.yaml (hunk 1 above)
Comment (ibuziuk) [reply deploy/helm/values.yaml:19]: yup, looks good

**Comment by ibuziuk** [comment]
@0marq also, please submit the Eclipse Contributor Agreement - https://www.eclipse.org/legal/ECA.php
This is a pre-requirement for getting the PRs merged

File: README.md (hunk 2 above)
Comment (0marq) [reply README.md]: Done

**Comment by 0marq** [comment]
@ibuziuk Everything looks good on my side ! Made the last changes to the README and submitted the Eclipse Contributor Agreement.

**Review by amisevsk — APPROVED** [review]
LGTM, thanks for the contribution @0marq!

**Review by ibuziuk — APPROVED** [review]

**Comment by l0rd** [comment]
@ibuziuk there is no new and noteworthy label in this repo. Should we create one?

**Comment by ibuziuk** [comment]
good point, added the label

**Pull request merged**